common/src/chain/transaction/mod.rs

```diff
@@ -143,6 +143,30 @@ impl Transaction {
     }
 }
 
+pub struct Transactions {
```

Comment:
What's the value of having this separate struct instead of just passing a Vec?

Reply:
I think I misunderstood this comment and ended up creating this new type https://github.com/mintlayer/mintlayer-core/pull/755#discussion_r1134585835

Yep, sounds good. I'll simplify this by passing in Vec.

Reply:
The idea from that comment is that there are two ways where we need to pass transactions to be included in a block:

1. Through an accumulator from the mempool
2. Through RPC, where we provide a Vec of transactions

Both should be one parameter unified through an enum, where choice one is used in normal operations, and choice two is used in RPC

Reply:
While I was reverting the `Transactions` struct, it wouldn't compile because I've made `make_block()` async which now makes `accumulator` possibly outlive the function. The reason for making it async is that `make_block()` will now contain all the mining work (via `finalize_consensus_data()`) by both methods (mempool and RPC), and can soon be taken out of the Tokio executor (not implemented yet) so that it doesn't hold up the rest of the system.

Reply:
Also, to simplify things further, I've stripped away the `Transaction` and the need for any enum by going straight for an array slice:

```
    pub async fn make_block(                  
        &self,                                
        current_tip_id: Id<Block>,            
        transactions: &[SignedTransaction],   
    ) -> Result<Block, BlockProductionError> {
```

Reply:
Ok, this ended up not working once I added the RPC call because the created block refers to the transactions that live longer. I'm going to commit my current rebased RPC changes to #755 and come back to this later - I don't want to be bogged down be one issue.